src/cljs/app/events.cljs

```diff
@@ -11,27 +11,49 @@
   [db [_ active-panel]]
   (assoc db :active-panel active-panel))
 
+(defn- navigate-to-url
+  [{:keys [db]} [_ url-string]]
+  {:db db
+   :change-url url-string})
+
 (defn- search-change-query
   [db [_ new-value]]
   (assoc-in db [:search :query] new-value))
 
-(defn- search-receive-results
+(defn- search-ragams-response
   [db [_ search-results]]
   (-> db
        (assoc-in [:search :results] search-results)
        (assoc :loading false)))
 
-(defn- search-ragams
+(defn- search-ragams-request
   [{:keys [db]} [_ query]]
   {:db (assoc db :loading true)
     :http-get {:endpoint "search"
-               :query query
-               :type "ragam"}})
+               :params {:query query
+                        :type "ragam"}
+               :on-success #(re-frame/dispatch [:search/receive-results %])}})
+
+(defn- get-ragam-request
+  [{:keys [db]} [_ ragam-id]]
+  {:db       (assoc-in db [:ragam :loading] true)
+   :http-get {:endpoint (str "ragam" "/" ragam-id)
+              :params nil
+              :on-success #(re-frame/dispatch [:ragam/receive-data %])}})
```

Comment:
Yeah, this could go to callback hell. Another way without using `core.async` is to not pass callbacks as arguments and add a separate effect for each callback. So, instead of `:http-get` it will be `:get-results`, or `:get-ragam` with the endpoint and on-success callback hardcoded into the effect. 

However, I'm not sure if I'd be violating `DRY` or something else then.